src/comgr.cpp

```diff
@@ -706,5 +718,46 @@ void BuildOcl(const std::string& name,
     }
 }
 
+void BuildAsm(const std::string& name,
+              const std::string& text,
+              const std::string& options,
+              const std::string& device,
+              std::vector<char>& binary)
+{
+    PrintVersion();
+    try
+    {
+        const Dataset inputs;
+        inputs.AddData(name, text, AMD_COMGR_DATA_KIND_SOURCE);
+
+        const ActionInfo action;
+        SetIsaName(action, device);
+        action.SetLogging(true);
+        auto optAsm = miopen::SplitSpaceSeparated(options);
+        compiler::lc::gcnasm::RemoveOptionsUnwanted(optAsm);
+        action.SetOptionList(optAsm);
+
+        const Dataset relocatable;
+        action.Do(AMD_COMGR_ACTION_ASSEMBLE_SOURCE_TO_RELOCATABLE, inputs, relocatable);
+
+        action.SetOptionList(OptionList());
+        const Dataset exe;
+        action.Do(AMD_COMGR_ACTION_LINK_RELOCATABLE_TO_EXECUTABLE, relocatable, exe);
+
+        constexpr auto INTENTIONALY_UNKNOWN = static_cast<amd_comgr_status_t>(0xffff);
+        if(exe.GetDataCount(AMD_COMGR_DATA_KIND_EXECUTABLE) < 1)
+            throw ComgrError{INTENTIONALY_UNKNOWN, "Executable binary not found"};
+        // Assume that the first exec data contains the binary we need.
+        const auto data = exe.GetData(AMD_COMGR_DATA_KIND_EXECUTABLE, 0);
+        data.GetBytes(binary);
+    }
+    catch(ComgrError& ex)
+    {
+        MIOPEN_LOG_E("comgr status = " << GetStatusText(ex.status));
+        if(!ex.text.empty())
+            MIOPEN_LOG_W(ex.text);
+    }
+}
+
 } // namespace comgr
```

Comment:
The comment needs to be updated

Reply:
Which one? This curly brace closes `miopen::comgr`